### PR TITLE
Ladder 4 editors field

### PR DESCRIPTION
**Pull request by lmcnulty**
Per #620,

- Adds an `editors` field to the `incidents` collection scheme
- Updates the MongoDB doc to include the new field
- Adds the field to those GraphQL queries and mutations in incidents.js that included every field.
- Adds a migration making Sean the editor for all incidents where one is not set.
- Adds the editor to the incidents table in `/cite` and replaces Sean in the citations with whoever the first editor is.
- In the incidents edit form, adds a field for editors
- Adds a test for the editor appearing in the incidents table.

### CODE REVIEW

File: site/realm/data_sources/mongodb-atlas/aiidprod/incidents/schema.json
Comment (smcgregor) [inline site/realm/data_sources/mongodb-atlas/aiidprod/incidents/schema.json:30]: I think there is a value that can go in the schema that will reject any changes that don't include the `editors` field. Let's set that so the database integrity is defended by the schema and we don't need to figure out why the build failed later.

**Review by smcgregor — COMMENTED** [review]
* You covered a lot of ground here, great job!
* I have a few inline comments that need changes or responses
* I think there will be a problem on ingestion of new incidents. It looks like a default editor is not being assigned. In the future we will tie this to user accounts, but for now it can be a tag people assign after the submission is promoted to an incident. Upon ingestion you can place my name as the default until that time.

**Needed in this PR**

* A default editor. Likely here: https://github.com/responsible-ai-collaborative/aiid/blob/master/site/realm/functions/promoteSubmissionToReport.js#L3 <--- this is a Realm function. It is essentially a lambda function that runs on the cloud on demand. It is responsible for mutating the database when the form for adding an incident is submitted.

**Future Issues (can be addressed after we merge this)**

* Tie the editor to the user that promotes the submission to a report

File: site/gatsby-site/src/components/BibTex.js
Comment (lmcnulty) [reply site/gatsby-site/src/components/BibTex.js:26]: That's generally my preference as well, but it does seem to contradict the linter. I wrapped those lines in `eslint-disable`, `eslint-enable`.

File: site/realm/data_sources/mongodb-atlas/aiidprod/incidents/schema.json
Comment (lmcnulty) [reply site/realm/data_sources/mongodb-atlas/aiidprod/incidents/schema.json:30]: In addition to "editors", I've added as required fields "_id", "incident_id", and "reports", those being the ones that wouldn't cause the schema validation to fail if required.

File: site/gatsby-site/cypress/fixtures/incidents/incident.json
Comment (lmcnulty) [reply site/gatsby-site/cypress/fixtures/incidents/incident.json]: Whoops. Thanks!

**Comment by lmcnulty** [comment]
I added `editors: ["Sean McGregor"]` to the incident object created in [promoteSubmissionToReport.js](https://github.com/responsible-ai-collaborative/aiid/pull/631/commits/c2355eb609dad3507159987a6607d331bc96bfc8#diff-567647866eb0f4fab30e4ddf5a54f965ee77ebba93396d7fc042df825699cdd7), which I expect to result in it being set on new incidents. However, I haven't been able to test this because I haven't figured out how to become an admin within my dev environment so that I promote a record to an incident.

**Comment by smcgregor** [comment]
Please let me know if this does not work, but I believe you go into the Realm UI, visit `App Users`, click the `Add New User` button.

Screen Shot 2022-05-20 at 4 41 39 PM

Then I think you should see a user here:

Screen Shot 2022-05-20 at 4 44 06 PM
And the `roles` array should contain an `admin` role.

File: site/gatsby-site/src/components/BibTex.js
Comment (smcgregor) [reply site/gatsby-site/src/components/BibTex.js:26]: Ah, got it. It is generally better to figure out how to make the linter work with you than to disable the linter. In this case it might be best to just be OK with the whitespace.

**Comment by smcgregor** [comment]
Second review is looking good. Let me know when I should do a final review and merge!

**Comment by lmcnulty** [comment]
Ok, so after making an admin account and trying to handle a sequence of errors that ultimately went away after running `realm-cli push` again, I'm still getting this runtime error when I press the "Add New Report" button:

> reason="role \\"role is admin\\" in \\"aiidprod.incidents\\" does not have insert permission: could not validate document: \n\t(root): _id is required"; code="SchemaValidationFailedWrite"; untrusted="insert not permitted"; details=map[]

That error goes away, and a new entry with the default editor is inserted, if in promoteSubmissionToReport.js I add:

```
const newIncident = {
   "_id": new BSON.ObjectId(),
   ...
}
```

But I don't understand why that's necessary. I thought that an `_id` was already supposed to be generated automatically – and that must be the case outside my dev environment, or else it wouldn't have worked so far for anyone else.

**Review by smcgregor — APPROVED** [review]

**Comment by smcgregor** [comment]
@cesarvarela merging this now, please feel free to be a reviewer/approver at your option.

**Pull request merged**
